dvc/repo/diff.py

```diff
@@ -0,0 +1,153 @@
+    diff_size = 0
+    for key in keys:
+        if key in a_entries and key in b_entries:
+            if a_entries[key] == b_entries[key]:
+                result["ident"] += 1
+            else:
+                result["changes"] += 1
+                diff_size += os.path.getsize(
+                    self.cache.local.get(b_entries[key])
+                ) - os.path.getsize(self.cache.local.get(a_entries[key]))
+        elif key in a_entries:
+            result["del"] += 1
+            diff_size -= os.path.getsize(self.cache.local.get(a_entries[key]))
+        else:
+            result["new"] += 1
+            diff_size += os.path.getsize(self.cache.local.get(b_entries[key]))
+    result["size_diff"] = diff_size
+    return result
+
+
+def _get_diff_outs(self, diff_dct):
+    self.tree = diff_dct["a_tree"]
+    a_outs = {str(out): out for st in self.stages() for out in st.outs}
+    self.tree = diff_dct["b_tree"]
+    b_outs = {str(out): out for st in self.stages() for out in st.outs}
+    outs_paths = set(a_outs.keys())
+    outs_paths.update(b_outs.keys())
+    results = {}
+    for path in outs_paths:
+        if path in a_outs and path in b_outs:
+            results[path] = {
+                "a_output": a_outs[path],
+                "b_output": b_outs[path],
+                "new_file": False,
+                "deleted_file": False,
+                # possible drawback: regular file ->directory movement
+                "is_dir": a_outs[path].is_dir_cache,
+            }
+        elif path in a_outs:
+            results[path] = {
+                "a_output": a_outs[path],
+                "new_file": False,
+                "deleted_file": True,
+                "is_dir": a_outs[path].is_dir_cache,
+            }
+        else:
+            results[path] = {
+                "b_output": b_outs[path],
+                "new_file": True,
+                "deleted_file": False,
+                "is_dir": b_outs[path].is_dir_cache,
+            }
+    return results
+
+
+def _diff_dir(self, target, diff_dct):
+    result = {"target": target}
+    result["is_dir"] = True
+    a_entries, b_entries = {}, {}
+    if not diff_dct["new_file"]:
+        result["old_file"] = target
+        result["old_checksum"] = diff_dct["a_output"].checksum
+        a_entries = _extract_dir(self, diff_dct["a_output"])
+    if not diff_dct["deleted_file"]:
+        result["new_file"] = target
+        result["new_checksum"] = diff_dct["b_output"].checksum
+        b_entries = _extract_dir(self, diff_dct["b_output"])
+    result.update(_get_tree_changes(self, a_entries, b_entries))
+    return result
+
+
+def _diff_file(self, target, diff_dct):
+    result = {"target": target}
+    size = 0
+    if not diff_dct["new_file"]:
+        result["old_file"] = target
+        result["old_checksum"] = diff_dct["a_output"].checksum
+        size -= os.path.getsize(
+            self.cache.local.get(diff_dct["a_output"].checksum)
+        )
+    print("first", size)
```

Comment:
repo.diff() should not print anything. All printing should be handled in dvc/command/diff.py

Reply:
it was debugging thing, should've deleted this, sorry for that